nats.go

```diff
@@ -350,7 +350,7 @@ type Conn struct {
 	// atomic.* functions crash on 32bit machines if operand is not aligned
 	// at 64bit. See https://github.com/golang/go/issues/599
 	Statistics
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
These are more complex than simple mutexes and unless there is a large amount of contention where reads outweigh writes substantially this is usually not worth it.